### ISSUE TITLE
Typescript formatting difference between Theia and VSCode

### ISSUE THREAD

**Issue by simark** [theia-ide/theia#1789]
When autoformatting Typescript source file, there is this difference:

Theia:

```
setTimeout(function() {
}, 1000);
```

VSCode:

```
setTimeout(function () {
}, 1000);
```

We should try to fix it, otherwise we'll always have annoying spurious changes if some people use Theia and others use VSCode.

**Comment by gorkem** [comment]
Which one is the truth result according to settings?

**Comment by simark** [comment]
> Which one is the truth result according to settings?

Who/what the is authority about Typescript formatting?

**Comment by simark** [comment]
Note that it would be quite complicated to get VSCode to change its behavior, and I don't think one is more right than the other, so we should probably just align Theia (well, typescript-language-server) on VSCode.

**Comment by gorkem** [comment]
I think the difference is because Theia does not have any preferences for typescript formatting. if Theia has the preferences that matches `typescript.format.*`settings keys of the vscode (which is the whole list of what tsserver has anyway) the difference will not be there. So this is more of a request to support formatting options on ts language server.

**Comment by vince-fugnitto** [comment]
closed for the moment thanks to #3757
